include/nuttx/spinlock.h

```diff
@@ -449,4 +454,98 @@ void spin_unlock_irqrestore_wo_note(FAR spinlock_t *lock, irqstate_t flags);
 #  define spin_unlock_irqrestore_wo_note(l, f) up_irq_restore(f)
 #endif
 
+#if defined(CONFIG_RW_SPINLOCK)
+
+/****************************************************************************
+ * Name: rw_read_spin_lock
```

Comment:
let's use the same name from https://elixir.bootlin.com/linux/latest/source/include/linux/rwlock.h:
rwlock_t, rwlock_init, read_trylock, write_trylock, read_lock, write_lock, read_unlock and write_unlock.